lib/rules/object-shorthand.js

```diff
@@ -248,11 +263,81 @@ module.exports = {
             return fixer.replaceTextRange([node.range[0], lastKeyToken.range[1]], `${keyText}: ${functionHeader}`);
         }
 
+        /*
+         * To determine whether a given arrow function has a lexical identifier (`this`, `arguments`, `super`, or `new.target`),
+         * create a stack of functions that define these identifiers (i.e. all functions except arrow functions) as the AST is
+         * traversed. Whenever a new function is encountered, create a new entry on the stack (corresponding to a different lexical
+         * scope of `this`, and whenever a function is exited, pop that entry off the stack. When an arrow function is entered,
```

Comment:
There is a closing parenthesis missing here somewhere. I assume after `scope of 'this'`?